src/components/containers/zoom-helpers.js

```diff
@@ -128,26 +128,32 @@ const Helpers = {
     return undefined;
   },
 
+  getLastDomain(targetProps, originalDomain) {
+    const { zoomDomain, cachedZoomDomain, currentDomain, domain } = targetProps;
+    if (zoomDomain && !isEqual(zoomDomain, cachedZoomDomain)) {
+      return defaults(
+        {}, zoomDomain, domain
+      );
+    }
+    return defaults(
+      {}, currentDomain || zoomDomain || originalDomain, domain
+    );
+  },
+
   getDomain(props) {
     const { originalDomain, domain, scale } = props;
     const scaleDomain = { x: scale.x.domain(), y: scale.y.domain() };
     return defaults({}, originalDomain, domain, scaleDomain);
   },
 
-  onMouseDown(evt, targetProps) {
+  onMouseDown(evt) {
     evt.preventDefault();
-    const { domain, zoomDomain } = targetProps;
-    const originalDomain = this.getDomain(targetProps);
-    const currentDomain = defaults(
-      {}, targetProps.currentDomain || zoomDomain || originalDomain, domain
-    );
     const { x, y } = Selection.getSVGEventCoordinates(evt);
     return [{
       target: "parent",
       mutation: () => {
         return {
```

Comment:
no reason why onMouseDown needs to squash values for the various `*domain` props. this makes the `this.getLastDomain` fix work.